sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/pooling/PooledStringBuilder.java

```diff
@@ -0,0 +1,184 @@
+package com.azure.cosmos.implementation.pooling;
+
+public class PooledStringBuilder {
+    final int MAX_SIZE = 1024;
+    StringBuilder builder = new StringBuilder();
+
+    ObjectPool<PooledStringBuilder> pool;
+
+    static ObjectPool<PooledStringBuilder> s_poolInstance = createPool();
+
+    PooledStringBuilder(ObjectPool<PooledStringBuilder> pool) {
+        this.pool = pool;
+    }
+
+    public static PooledStringBuilder createInstance() {
+        return s_poolInstance.get();
+    }
+    public static ObjectPool<PooledStringBuilder> createPool() {
+        return createPool(32);
+    }
+
+    public static ObjectPool<PooledStringBuilder> createPool(int size) {
+        return new ObjectPool<PooledStringBuilder>((pool) -> new PooledStringBuilder(pool), size);
```

Comment:
We are creating a new `ObjectFactory` every single time. Why can't we use the same factory instance?
What is the limit on number of `ObjectFactory` instances that can be created?